OCP 4.16 Release Notes for P/Z (#76393)
openshift/openshift-docs: release_notes/ocp-4-16-release-notes.adoc

.{product-title} features
[cols="3,1,1",options="header"]
|====
|Feature |{ibm-power-name} |{ibm-z-name} and {ibm-linuxone-name}

|Alternate authentication providers
|Supported
|Supported

|Agent-based Installer
|Supported
|Supported

|Assisted Installer
|Supported
|Supported

|Automatic Device Discovery with Local Storage Operator
|Unsupported
|Supported

|Automatic repair of damaged machines with machine health checking
|Unsupported
|Unsupported

|Cloud controller manager for {ibm-cloud-name}
|Supported
|Unsupported

|Controlling overcommit and managing container density on nodes
|Unsupported
|Unsupported

|Cron jobs
|Supported
|Supported

|Descheduler
|Supported
|Supported

|Egress IP
|Supported
|Supported

|Encrypting data stored in etcd
|Supported
|Supported

|FIPS cryptography
|Supported
|Supported

|Helm
|Supported
|Supported

|Horizontal pod autoscaling
|Supported
|Supported

|Hosted control planes (Technology Preview)
|Supported
|Supported

|IBM Secure Execution
|Unsupported
|Supported

> |{ibm-power-server-name} Block CSI Driver Operator
> |Supported
> |Unsupported

Reviewer: @alishaIBM I think this should be moved to the Operators table

```suggestion
|{ibm-power-server-name} Block CSI Driver Operator
|Supported
|Unsupported
```

Author: Agree. Moved to the `Operators` section. Thanks.

|Installer-provisioned Infrastructure Enablement for {ibm-power-server-name}
|Supported
|Unsupported

|Installing on a single node
|Supported
|Supported

|IPv6
|Supported
|Supported

|Monitoring for user-defined projects
|Supported
|Supported

|Multi-architecture compute nodes
|Supported
|Supported

|Multi-architecture {ibm-power-name} control plane with support of Intel and {ibm-power-name} compute nodes
|Supported
|Supported

|Multipathing
|Supported
|Supported

|Network-Bound Disk Encryption - External Tang Server
|Supported
|Supported

|Non--volatile memory express drives (NVMe)
|Supported
|Unsupported

|nx-gzip for Power10 (Hardware Acceleration)
|Supported
|Unsupported

|oc-mirror plugin
|Supported
|Supported

|OpenShift CLI (`oc`) plugins
|Supported
|Supported

|Operator API
|Supported
|Supported

|OpenShift Virtualization
|Unsupported
|Unsupported

|OVN-Kubernetes, including IPsec encryption
|Supported
|Supported

|PodDisruptionBudget
|Supported
|Supported

|Precision Time Protocol (PTP) hardware
|Unsupported
|Unsupported

|{openshift-local-productname}
|Unsupported
|Unsupported

|Scheduler profiles
|Supported
|Supported

> |Stream Control Transmission Protocol (SCTP)

Author: Updated the PR as per above suggestion. Thanks @SNiemann15

[id="ocp-4-16-ibm-z_{context}"]
=== {ibm-z-title} and {ibm-linuxone-title}

With this release, {ibm-z-name} and {ibm-linuxone-name} are now compatible with {product-title} {product-version}. You can perform the installation with z/VM, LPAR, or {op-system-base-full} Kernel-based Virtual Machine (KVM). For installation instructions, see the following documentation:

* xref:../installing/installing_ibm_z/preparing-to-install-on-ibm-z.adoc#preparing-to-install-on-ibm-z[Installing a cluster with on {ibm-z-title} and {ibm-linuxone-title}]

[IMPORTANT]
====
Compute nodes must run {op-system-first}.
====

[discrete]
==== {ibm-z-title} and {ibm-linuxone-title} notable enhancements

The {ibm-z-name} and {ibm-linuxone-name} release on {product-title} {product-version} adds improvements and new capabilities to {product-title} components and concepts.

This release introduces support for the following features on {ibm-z-name} and {ibm-linuxone-name}:

* Agent-based Installer ISO boot

> * Ingress Node Firewall Operator

Reviewer: ```suggestion
* Ingress Node Firewall Operator
* Multi-architecture compute machines in an LPAR
```

Author: Ack. Updated the PR.

> * Secure boot

Reviewer: ```suggestion
* Secure boot
* Installing on a single node in an LPAR
```

[id="ocp-4-16-ibm-z_{context}"]
=== {ibm-z-title} and {ibm-linuxone-title}

> With this release, {ibm-z-name} and {ibm-linuxone-name} are now compatible with {product-title} {product-version}. You can perform the installation with z/VM, LPAR, or {op-system-base-full} Kernel-based Virtual Machine (KVM). For installation instructions, see the following documentation:

Reviewer 2: You can include the link right in the sentence.

```suggestion
With this release, {ibm-z-name} and {ibm-linuxone-name} are now compatible with {product-title} {product-version}. You can perform the installation with z/VM, LPAR, or {op-system-base-full} Kernel-based Virtual Machine (KVM). For installation instructions, see xref:../installing/installing_ibm_z/preparing-to-install-on-ibm-z.adoc#preparing-to-install-on-ibm-z[Installing a cluster with on {ibm-z-title} and {ibm-linuxone-title}]
```

> * xref:../installing/installing_ibm_z/preparing-to-install-on-ibm-z.adoc#preparing-to-install-on-ibm-z[Installing a cluster with on {ibm-z-title} and {ibm-linuxone-title}]

Reviewer 2: If you add the link to the sentence, remove this line.

This release introduces support for the following features on {ibm-z-name} and {ibm-linuxone-name}:

* Agent-based Installer ISO boot
* Ingress Node Firewall Operator
* Secure boot

[id="ocp-4-16-ibm-power_{context}"]
=== {ibm-power-title}

{ibm-power-name} is now compatible with {product-title} {product-version}. For installation instructions, see the following documentation:

* xref:../installing/installing_ibm_power/installing-ibm-power.adoc#installing-ibm-power_installing-ibm-power[Installing a cluster on {ibm-power-name}]
* xref:../installing/installing_ibm_power/installing-restricted-networks-ibm-power.adoc#installing-restricted-networks-ibm-power_installing-restricted-networks-ibm-power[Installing a cluster on {ibm-power-name} in a restricted network]

[IMPORTANT]
====
Compute nodes must run {op-system-first}.
====

[discrete]
==== {ibm-power-title} notable enhancements

The {ibm-power-name} release on {product-title} {product-version} adds improvements and new capabilities to {product-title} components.

This release introduces support for the following features on {ibm-power-name}:

* CPU manager
* Ingress Node Firewall Operator

> * Installer-provisioned Infrastructure for IBM Power® Virtual Server - move to CAPI

Reviewer 3: @alishaIBM I think it needs one minor update to remove `-move to CAPI` I need mjurek to confirm.

> * Installer-provisioned Infrastructure for IBM Power® Virtual Server - move to CAPI

Reviewer 4: I think we remove the entire IPI line, we GA'd in 4.15 and are no longer moving to CAPI

> * Installer-provisioned Infrastructure for IBM Power® Virtual Server - move to CAPI

Author: Ack. Done.

|Supported
|Unsupported

|Installing on a single node
|Supported
|Supported

|IPv6
|Supported
|Supported

|Monitoring for user-defined projects
|Supported
|Supported

|Multi-architecture compute nodes
|Supported
|Supported

|Multi-architecture control plane
|Supported
|Supported

|Multipathing
|Supported
|Supported

|Network-Bound Disk Encryption - External Tang Server
|Supported
|Supported

|Non--volatile memory express drives (NVMe)
|Supported
|Unsupported

|nx-gzip for Power10 (Hardware Acceleration)
|Supported
|Unsupported

|oc-mirror plugin
|Supported
|Supported

|OpenShift CLI (`oc`) plugins
|Supported
|Supported

|Operator API
|Supported
|Supported

|OpenShift Virtualization
|Unsupported
|Unsupported

|OVN-Kubernetes, including IPsec encryption
|Supported
|Supported

|PodDisruptionBudget
|Supported
|Supported

|Precision Time Protocol (PTP) hardware
|Unsupported
|Unsupported

|{openshift-local-productname}
|Unsupported
|Unsupported

|Scheduler profiles
|Supported
|Supported

|Secure Boot
|Unsupported
|Supported

|Stream Control Transmission Protocol (SCTP)
|Supported
|Supported

|Support for multiple network interfaces
|Supported
|Supported

> |The openshift-install utility to support various SMT levels on {ibm-power-name} (Hardware Acceleration)

Reviewer 2: ```suggestion
|The `openshift-install` utility to support various SMT levels on {ibm-power-name} (Hardware Acceleration)
```

[id="ocp-4-16-ibm-z_{context}"]
=== {ibm-z-title} and {ibm-linuxone-title}

With this release, {ibm-z-name} and {ibm-linuxone-name} are now compatible with {product-title} {product-version}. You can perform the installation with z/VM, LPAR, or {op-system-base-full} Kernel-based Virtual Machine (KVM). For installation instructions, see the following documentation:

* xref:../installing/installing_ibm_z/preparing-to-install-on-ibm-z.adoc#preparing-to-install-on-ibm-z[Installing a cluster with on {ibm-z-title} and {ibm-linuxone-title}]

[IMPORTANT]
====
Compute nodes must run {op-system-first}.
====

[discrete]
==== {ibm-z-title} and {ibm-linuxone-title} notable enhancements

The {ibm-z-name} and {ibm-linuxone-name} release on {product-title} {product-version} adds improvements and new capabilities to {product-title} components and concepts.

This release introduces support for the following features on {ibm-z-name} and {ibm-linuxone-name}:

> * Agent-based Installer ISO boot

Reviewer: ```suggestion
* Agent-based Installer ISO boot for {op-system-base} KVM
```
```suggestion
* Agent-based Installer ISO boot
```

* Ingress Node Firewall Operator

> * Installing on a single node in an LPAR

Reviewer: ```suggestion

```
```suggestion
* Installing on a single node in an LPAR
```

Author: Ack. Removed line 131.

* Multi-architecture compute machines in an LPAR

> * Secure boot

Author: Ack. Updated the PR. cc: @SNiemann15

[id="ocp-4-16-ibm-z_{context}"]
=== {ibm-z-title} and {ibm-linuxone-title}

With this release, {ibm-z-name} and {ibm-linuxone-name} are now compatible with {product-title} {product-version}. You can perform the installation with z/VM, LPAR, or {op-system-base-full} Kernel-based Virtual Machine (KVM). For installation instructions, see 

> xref:../installing/installing_ibm_z/preparing-to-install-on-ibm-z.adoc#preparing-to-install-on-ibm-z[Installing a cluster with on {ibm-z-title} and {ibm-linuxone-title}]

Reviewer 5: ```suggestion
xref:../installing/installing_ibm_z/preparing-to-install-on-ibm-z.adoc#preparing-to-install-on-ibm-z[Installing a cluster with on {ibm-z-title} and {ibm-linuxone-title}].
```
A complete sentence; needs a period.

Author: Ack. Updated the PR.

[id="ocp-4-16-ibm-z_{context}"]
=== {ibm-z-title} and {ibm-linuxone-title}

With this release, {ibm-z-name} and {ibm-linuxone-name} are now compatible with {product-title} {product-version}. You can perform the installation with z/VM, LPAR, or {op-system-base-full} Kernel-based Virtual Machine (KVM). For installation instructions, see 
xref:../installing/installing_ibm_z/preparing-to-install-on-ibm-z.adoc#preparing-to-install-on-ibm-z[Installing a cluster with on {ibm-z-title} and {ibm-linuxone-title}].

[IMPORTANT]
====
Compute nodes must run {op-system-first}.
====

> [discrete]
> ==== {ibm-z-title} and {ibm-linuxone-title} notable enhancements

Reviewer 6: Per our [docs guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#discrete-headings), even if a heading is discrete, it still needs a unique ID: 
```suggestion
[discrete]
[id="ocp-4-16-ibm-z-enhancements_{context}"]
==== {ibm-z-title} and {ibm-linuxone-title} notable enhancements
```

The {ibm-z-name} and {ibm-linuxone-name} release on {product-title} {product-version} adds improvements and new capabilities to {product-title} components and concepts.

This release introduces support for the following features on {ibm-z-name} and {ibm-linuxone-name}:

* Agent-based Installer ISO boot for {op-system-base} KVM
* Ingress Node Firewall Operator
* Multi-architecture compute machines in an LPAR
* Secure boot for z/VM and LPAR

[id="ocp-4-16-ibm-power_{context}"]
=== {ibm-power-title}

{ibm-power-name} is now compatible with {product-title} {product-version}. For installation instructions, see the following documentation:

* xref:../installing/installing_ibm_power/installing-ibm-power.adoc#installing-ibm-power_installing-ibm-power[Installing a cluster on {ibm-power-name}]
* xref:../installing/installing_ibm_power/installing-restricted-networks-ibm-power.adoc#installing-restricted-networks-ibm-power_installing-restricted-networks-ibm-power[Installing a cluster on {ibm-power-name} in a restricted network]

[IMPORTANT]
====
Compute nodes must run {op-system-first}.
====

> [discrete]
> ==== {ibm-power-title} notable enhancements

Reviewer 6: Same comment as above:
```suggestion
[discrete]
[id="ocp-4-16-ibm-power-enhancements_{context}"]
==== {ibm-power-title} notable enhancements
```

The {ibm-power-name} release on {product-title} {product-version} adds improvements and new capabilities to {product-title} components.

This release introduces support for the following features on {ibm-power-name}:

* CPU manager
* Ingress Node Firewall Operator

> [discrete]
> === {ibm-power-title}, {ibm-z-title}, and {ibm-linuxone-title} support matrix

Reviewer 6: Same comment as above:
```suggestion
[discrete]
[id="ocp-4-16-ibm-z-power-support-matrix_{context}"]
=== {ibm-power-title}, {ibm-z-title}, and {ibm-linuxone-title} support matrix
```
